packages/react-dom/src/events/DOMModernPluginEventSystem.js

```diff
@@ -165,6 +167,57 @@ export const reactScopeListenerStore: WeakMap<
   >,
 > = new PossiblyWeakMap();
 
+function executeDispatch(
+  event: ReactSyntheticEvent,
+  listener: Function,
+  currentTarget: EventTarget,
+): void {
+  const type = event.type || 'unknown-event';
+  event.currentTarget = currentTarget;
+  invokeGuardedCallbackAndCatchFirstError(type, listener, undefined, event);
+  event.currentTarget = null;
+}
+
+function executeDispatchesInOrder(event: ReactSyntheticEvent): void {
+  // TODO we should remove _dispatchListeners and _dispatchInstances at some point.
+  const dispatchListeners = event._dispatchListeners;
+  const dispatchInstances = event._dispatchInstances;
+  let previousInstance;
+
+  if (dispatchListeners !== null && dispatchInstances !== null) {
+    for (let i = 0; i < dispatchListeners.length; i++) {
+      const dispatchInstance = dispatchInstances[i];
+      const dispatchListener = dispatchListeners[i];
+      let instance = dispatchInstance;
+      let currentTarget;
+
+      // dispatchInstance can come in two forms:
+      // - as a Fiber instance
+      // - as an object { instance: Fiber | null, container: EventTarget }
+      // The second form is used for useEvent types
+      if (dispatchInstance.container !== undefined) {
+        instance = ((dispatchInstance: any): DispatchInstance).instance;
+        currentTarget = ((dispatchInstance: any): DispatchInstance).container;
+      } else {
+        currentTarget = getNodeFromInstance(((dispatchInstance: any): Fiber));
+      }
+      // We check if the instance was the same as the last one,
+      // if it was, then we're still on the same instance thus
```

Comment:
What does it mean to be on the same instance? Several `useEvent` targeting the same node?

Reply:
Exactly, it's for `useEvent` targeting the same node.